sale_zero_stock_approval/models/sale_order.py

```diff
@@ -0,0 +1,22 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    zero_stock_approval = fields.Boolean(
+        string="Approval"
+    )
+
+    @api.model
+    def fields_get(self, allfields=None, attributes=None):
+        res = super().fields_get(allfields, attributes)
+        if not self.env.user.has_group("sales_team.group_sale_manager") and "zero_stock_approval" in res:
+                res["zero_stock_approval"]["readonly"] = True
+        return res
+
+    def action_confirm(self):
+        if not self.zero_stock_approval and not self.env.user.has_group('sales_team.group_sale_manager'):
+            raise UserError('Sales order cannot be confirmed unless it is approved')
```

Comment:
```suggestion
            raise UserError(_('Sales order cannot be confirmed unless it is approved'))
```
its good to use _() for translation support

Reply:
done